examples/auth/app/components/Form.tsx

```diff
@@ -3,26 +3,26 @@ import {Form as FinalForm, FormProps as FinalFormProps} from "react-final-form"
 import * as z from "zod"
 export {FORM_ERROR} from "final-form"
 
-type FormProps<FormValues> = {
+type FormProps<S extends z.ZodType<any, any>> = {
   /** All your form fields */
   children: ReactNode
   /** Text to display in the submit button */
   submitText: string
-  onSubmit: FinalFormProps<FormValues>["onSubmit"]
-  initialValues?: FinalFormProps<FormValues>["initialValues"]
-  schema?: z.ZodType<any, any>
+  onSubmit: FinalFormProps<z.infer<S>>["onSubmit"]
+  initialValues?: FinalFormProps<z.infer<S>>["initialValues"]
+  schema?: S
 } & Omit<PropsWithoutRef<JSX.IntrinsicElements["form"]>, "onSubmit">
 
-export function Form<FormValues extends Record<string, unknown>>({
+export function Form<S extends z.ZodType<any, any>>({
   children,
   submitText,
   schema,
```

Comment:
Which one is more type-checked?

`ZodType<any, any>` `Record<string, unknown>`


It seems that unknown is better than any

Reply:
Thanks for reviewing!

In this case `S` is **extending** `z.ZodType<any, any>` which means that it's up to the user to define what the actual type will be. So it doesn't matter what we use here.

Reply:
I see! I'll try it next time. Thanks for the comment!